### DISCUSSION TITLE
Consider warning when editing to prevent users from thinking things aren't working properly due to permissions

### DISCUSSION THREAD

**Discussion by vertigo220** [FossifyOrg/General-Discussion#396] [Ideas]
I recently had issues with trying to edit images as when I tried to edit them in other apps, the apps would load but not the images. This happened with a few different apps, most of the ones I have with editing capabilities. I initially assumed it was a bug with SMT/Fossify Gallery since it was happening with multiple apps, yet opening them with Gallery worked just fine. It was only after I looked more closely at the error given by one of the apps (the only one to provide an error as opposed to just loading a blank canvas) that I realized it was a permissions issue, and all those apps that didn't work hadn't been granted the necessary storage permissions. While this is obviously not an issue with Gallery, and not technically its/your responsibility, it can nonetheless be a poor user experience and may lead users to believe it's Gallery's fault (as I did for a while). This could result in them reporting it as a bug or, worse, uninstalling Gallery and writing it off.

I propose some sort of warning to alert users that if the image doesn't load properly in the editing app they choose, they should check that app's permissions. Ideally, this would be done once per app, the first time it's chosen to open an image, though I'm not sure if that's possible, as it would require Gallery to know which app was chosen, and I'm not sure if it's given that info or if Android handles that. Also, that wouldn't help if they were notified once, then didn't use the app for a while and Android revoked its permissions, and then they tried it again later and found it not working. To help with that potential issue, Gallery could warn the first time the app is selected within a certain time span, e.g. however long Android typically revokes permissions. So if it revokes after an app is unused for a month, then if the app hasn't been selected for editing in over a month, it would warn.

If that's not possible, it could warn when the edit command is issued, i.e. when the user taps the edit icon or edit from the menu. Of course, to not be annoying, it would only do this once, so in the future the issue may be encountered due to using a different app or Android revoking the permissions, and by then the user may have forgotten this issue.

It's unfortunate steps like these are necessary due to poor permission implementation by Google, but that's the sad state of Android these days, and so to prevent users from having frustrating experiences and possibly discounting the app because of it, I do think such actions are recommended and even necessary. But I thought it should probably be a discussion to determine if others feel the same way and to decide on the best method before creating an issue.

**Comment by naveensingh** [comment]
This should be resolved in version 1.4.0. Editing should work properly as long as the editor app isn't barebones. See https://github.com/FossifyOrg/Gallery/issues/525 for more info.
